src/layers/core/geojson-layer/geojson-layer.js

```diff
@@ -81,23 +78,22 @@ export default class GeoJsonLayer extends CompositeLayer {
   }
 
   renderLayers() {
-    const {subLayers: {pointFeatures, lineFeatures, polygonFeatures,
-      polygonOutlineFeatures}} = this.state;
-    const {id, getPointColor, getPointSize, getStrokeColor, getStrokeWidth,
-      getFillColor, getElevation} = this.props;
-    const {extruded, wireframe} = this.props;
-
-    let {drawPoints, drawLines, drawPolygons, fillPolygons} = this.props;
-    drawPoints = drawPoints && pointFeatures && pointFeatures.length > 0;
-    drawLines = drawLines && lineFeatures && lineFeatures.length > 0;
-    drawPolygons = drawPolygons && polygonOutlineFeatures && polygonOutlineFeatures.length > 0;
-    fillPolygons = fillPolygons && polygonFeatures && polygonFeatures.length > 0;
+    const {features} = this.state;
+    const {pointFeatures, lineFeatures, polygonFeatures, polygonOutlineFeatures} = features;
+    const {getColor, getFillColor, getRadius, getStrokeWidth, getElevation} = this.props;
+    const {id, stroked, filled, extruded, wireframe} = this.props;
+
+    let {} = this.props;
```

Comment:
I don't particularly like this syntax in javascript... readers have no idea what variables are assigned here.